### DISCUSSION TITLE
SEO meeting URL's for TSML UI?

### DISCUSSION THREAD

**Discussion by intuitivecat** [code4recovery/12-step-meeting-list#1344] [Q&A] [answered]
I really like TSML_UI, but I also like the slugs being part of the rewrite_URL so we can promote the many (international) meetings we list in search results. 

TSML_UI: https://SITE/meetings/?meeting=meeting-slug
LEGACY:  https://SITE/meetings/meeting-slug/

Do I have this right, that TSML_UI cannot do the rewrite slugs?   Anyone have ideas for work-arounds?

**Comment by joshreisner** [comment, marked as answer]
that's true, but i think we can probably manage it using the same technique we used for online intergroup with their meeting finder

created an issue: https://github.com/code4recovery/tsml-ui/issues/357

**Reply by intuitivecat** [reply]
Thank you so much for the speedy feedback and sample code which I may play with. 

I am currently working on a project that involves 1083 meetings (Refuge Recovery, Recovery Dharma plus BuddhistRecovery.org independent listings)

So far I am finding that TSML is working great to aggregate like this.  

We do not want to lose our SEO rankings we have today, mostly for our region and subregion pages.  There I am finding that building out a simple custom plugin to render meetings by region/subregion is not that hard at all, but in the case of this issue it would force me not to use the shiny new TSML UI!

While I considered (and would consider) the rewrite approach for region page and the existing filter, I would like a bit more control over what is listed.   I feel it is better for SEO if there is a brief summary of the meeting on the list page as just a list of links is light on content, much like our listing pages today.
